src/backend/src/services/projects.services.ts

```diff
@@ -605,6 +605,112 @@ export default class ProjectsService {
     ).map(linkTypeTransformer);
   }
 
+  /**
+   * Creates a new Material
+   * @param creator the user creating the material
+   * @param name the name of the material
+   * @param assemblyId the id of the Assembly for the material
+   * @param status the Material Status of the material
+   * @param materialTypeName the name of the Material Type
+   * @param manufacturerName the name of the material's manufacturer
+   * @param manufacturerPartNumber the manufacturer part number for the material
+   * @param pdmFileName the name of the pdm file for the material
+   * @param quantity the quantity of material as a number
+   * @param unitName the name of the Quantity Unit the quantity is measured in
+   * @param price the price of the material in whole cents
+   * @param subtotal the subtotal of the price for the material in whole cents
+   * @param linkUrl the url for the material's link as a string
+   * @param notes any notes about the material as a string
+   * @param wbsNumber the WBS number of the WBS element associated with this material
+   * @returns the id of the created material
+   */
+  static async createMaterial(
+    creator: User,
+    name: string,
+    assemblyId: string,
+    status: Material_Status,
+    materialTypeName: string,
+    manufacturerName: string,
+    manufacturerPartNumber: string,
+    pdmFileName: string,
+    quantity: number,
+    unitName: string,
+    price: number,
+    subtotal: number,
+    linkUrl: string,
+    notes: string,
+    wbsNumber: WbsNumber
+  ): Promise<string> {
+    const wbsElement = await prisma.wBS_Element.findFirst({
+      where: {
+        carNumber: wbsNumber.carNumber,
+        projectNumber: wbsNumber.projectNumber,
+        workPackageNumber: wbsNumber.workPackageNumber
+      },
+      include: { project: { ...projectQueryArgs }, workPackage: { include: { project: { ...projectQueryArgs } } } }
+    });
+    if (!wbsElement) throw new NotFoundException('WBS Element', wbsPipe(wbsNumber));
+
+    const assembly = await prisma.assembly.findFirst({
+      where: { assemblyId }
+    });
+    if (assemblyId && !assembly) throw new NotFoundException('Assembly', assemblyId);
```

Comment:
this whole assembly check should just be wrapped in an if(assemblyId) {}, if the user isn't associating the material with an assembly we don't need to check anything about it